sound/soc/sof/sof-audio.h

```diff
@@ -246,6 +246,42 @@ int sof_set_hw_params_upon_resume(struct device *dev);
 bool snd_sof_stream_suspend_ignored(struct snd_sof_dev *sdev);
 bool snd_sof_dsp_only_d0i3_compatible_stream_active(struct snd_sof_dev *sdev);
 
+/*
+ * Helper function to determine the target DSP state during
+ * system suspend. This function only cares about the device
+ * D-states. Platform-specific substates, if any, should be
+ * handled by the platform-specific parts.
+ */
+static inline u32 snd_sof_dsp_power_target(struct snd_sof_dev *sdev)
```

Comment:
I'd keep this function in pm.c and modify the signature for snd_sof_release_trace(struct snd_sof_dev *sdev, bool release) to pass true/false for releasing. and it should be false when entering D0i3 and true otherwise

Reply:
Do you mean make this function as external function? I'm OK with it. I don't understand "pass true/false". Do you mean make the snd_sof_dsp_power_target() return true/false?
This change is recommended by @ujfalusi . What's do you think, @ujfalusi ?

Reply:
> Do you mean make this function as external function? I'm OK with it. I don't understand "pass true/false". Do you mean make the snd_sof_dsp_power_target() return true/false? This change is recommended by @ujfalusi . What's do you think, @ujfalusi ?

I don't think I have suggested to move the `snd_sof_dsp_power_target()`, if I recall I was asking for you to pass the target power state directly to release_trace() so that it can decide to not send the IPC if the DSP is going to remain enabled.

Reply:
You can pass the `pm_state` to the `snd_sof_release_trace()` which is also sent to the SOF clients if they are interested.

Reply:
@ujfalusi @ranj063 After a second thought, if we pass ```bool release``` to the function snd_sof_release_trace(), we don't need export ```snd_sof_dsp_power_target()```. snd_sof_release_trace() is called in pm.c except in snd_sof_free_trace(), which we have definite 'release' value. So I will follow @ranj063 suggestion

Reply:
> @ujfalusi @ranj063 After a second thought, if we pass `bool release` to the function snd_sof_release_trace(), we don't need export `snd_sof_dsp_power_target()`. snd_sof_release_trace() is called in pm.c except in snd_sof_free_trace(), which we have definite 'release' value. So I will follow @ranj063 suggestion

No, we should pass the `pm_state` to the `snd_sof_release_trace()`

Reply:
Agree with @ujfalusi , please pass pm_state to snd_sof_release_trace() and keep snd_sof_dsp_power_target() in pm.c

Reply:
@libinyang  @ujfalusi  What happened to passing "pm_state" to snd_sof_release_trace() ? I'm not fond of passing a boolean, it's not very readable. If we want to keep the target PM state out of trace.c, I'd then rather have a separate "snd_sof_stop_trace()" call (and snd_sof_release_trace() could call that internally). That would make the code more readable.

Reply:
> @libinyang @ujfalusi What happened to passing "pm_state" to snd_sof_release_trace() ? I'm not fond of passing a boolean, it's not very readable. If we want to keep the target PM state out of trace.c, I'd then rather have a separate "snd_sof_stop_trace()" call (and snd_sof_release_trace() could call that internally). That would make the code more readable.

@kv2019i, we should not pass boolean from pm.c to trace.c, I proposed to @libinyang that we should have a new `snd_sof_trace_suspend(sdev, pm_state);` and `snd_sof_trace_resume(sdev);` as an external interface. Internally we can pass around boolean, which we might need to to do the right thing in case of module removal.